### DISCUSSION TITLE
CycloneDDS configuration: Create3 + RPi4 + Other PCs

### DISCUSSION THREAD

**Discussion by royito55** [iRobotEducation/create3_docs#366] [Questions]
Hi,

I have 4 Create3 robots. I have connected one to a Raspberry Pi 4 following these instructions: https://iroboteducation.github.io/create3_docs/setup/pi4/

Since I want to ensure proper communication, I want to have the RPi4 connected to the Create3 through the USB-C cable, so I followed the steps to add `usb0` to the ethernets network_config, and created a cyclonedds.xml configuration file to indicate two network interface addresses: `usb0` and `wlan0`.

However, this setup prevents from discovery in another PC (apart from the RPi4) which renders the robot useless in a larger network. I realize I should also create an XML config file for the PC that wants to see the Create3 topics, but so far I've had no luck.

Is there anyone that has done something similar to this?

Here is my XML (I've commented the Discovery part because I don't think this part is relevant although it is right below the usb0 instructions in https://iroboteducation.github.io/create3_docs/setup/xml-config/#cyclonedds)

```
<CycloneDDS>
	<Domain>
		<General>
			<NetworkInterfaceAddress>usb0,wlan0</NetworkInterfaceAddress>
			<DontRoute>true</DontRoute>
		</General>
		<!--<Discovery>
			<MaxAutoParticipantIndex>100</MaxAutoParticipantIndex>
			<ParticipantIndex>auto</ParticipantIndex>
			<Peers>
				<Peer Address="${ROBOT_IP}:7410"/>
			</Peers>
		</Discovery>-->
	</Domain>
</CycloneDDS>

```

With this config, I can control the Create3 through the attached RPi4, BUT I can't see the Create3 topics from another PC. What should the XML config file be in that case?

**Comment by martinerk0** [comment]
@royito55 have you managed to see any topics from other PC using CycloneDDS?
